src/backend/core/models.py

```diff
@@ -158,6 +171,116 @@ def clean(self):
             raise exceptions.ValidationError({"data": [error_message]}) from e
 
 
+class OrganizationManager(models.Manager):
+    """
+    Custom manager for the Organization model, to manage complexity/automation.
+    """
+
+    def get_or_create(self, siret: str = None, domain: str = None, **kwargs):
+        """
+        Get or create an organization.
+
+        We expect to have only one organization per SIRET, but the SIRET might not be provided.
+        When the SIRET is not provided, we use the domain to identify the organization.
+        If both are provided, we use the SIRET first.
+        """
+        if not any([siret, domain]):
+            raise ValueError("You must provide either a siret or a domain.")
+
+        filters = models.Q()
+        if siret:
+            filters |= models.Q(sirets__icontains=siret)
+        if domain:
+            filters |= models.Q(domains__icontains=domain)
+
+        try:
+            organization = self.get(filters, **kwargs)
+            # If there are several organizations, we must raise an error and fix the data
+        except self.model.DoesNotExist:
+            organization = None
+
+        if organization:
+            return organization, False
+
+        # Manage the case where the organization does not exist
+        if siret:
+            return self.create(name=siret, sirets=[siret], **kwargs), True
+
+        if domain:
+            return self.create(name=domain, domains=[domain], **kwargs), True
+
+        raise ValueError("Should never reach this point.")
+
+
+def validate_unique_siret(value):
+    """
+    Validate that the SIRET values in an array field are unique across all instances.
+    """
+    if Organization.objects.filter(sirets__overlap=value).exists():
+        raise ValidationError("sirets value must be unique across all instances.")
+
+
+def validate_unique_domain(value):
+    """
+    Validate that the domain values in an array field are unique across all instances.
+    """
+    if Organization.objects.filter(domains__overlap=value).exists():
+        raise ValidationError("domains value must be unique across all instances.")
+
+
+class Organization(BaseModel):
+    """
+    Organization model used to regroup Teams.
+
+    Each User have an Organization, which corresponds actually to a default organization
+    because a user can belong to a Team from another organization.
+    Each Team have an Organization, which is the Organization from the User who created
+    the Team.
+
+    Organization is managed automatically, the User should never choose their Organization.
+    When creating a User, you must use the `get_or_create` method from the
+    OrganizationManager to find the proper Organization.
+    """
+
+    name = models.CharField(_("name"), max_length=100)
+    sirets = ArrayField(
```

Comment:
Can you say again in the docstring why we can have several sirets or domains for one organization?